apps/bixarena/api-gateway/src/main/java/org/sagebionetworks/bixarena/api/gateway/configuration/SecurityConfiguration.java

```diff
@@ -25,33 +20,29 @@
 @RequiredArgsConstructor
 public class SecurityConfiguration {
 
-  // private final AnonymousAccessGatewayFilter anonymousAccessFilter;
-  // private final ApiKeyAuthenticationGatewayFilter apiKeyAuthenticationFilter;
-  // private final JwtAuthenticationGatewayFilter jwtAuthenticationFilter;
+  private final RouteConfigRegistry routeConfigRegistry;
 
   @Bean
   SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http) {
-    return http
+    // Dynamically collect all anonymous endpoints from the route config registry
+    var anonymousPaths = routeConfigRegistry.getAnonymousPaths();
+
+    http
       .csrf(ServerHttpSecurity.CsrfSpec::disable)
-      // Add our custom authentication filters in order:
-      // 1. Anonymous access (generates JWTs for public endpoints)
-      // 2. API key authentication (exchanges API keys for JWTs)
-      // 3. JWT validation (validates existing JWTs)
-      // .addFilterBefore(anonymousAccessFilter, SecurityWebFiltersOrder.AUTHORIZATION)
-      // .addFilterBefore(apiKeyAuthenticationFilter, SecurityWebFiltersOrder.AUTHORIZATION)
-      // .addFilterBefore(jwtAuthenticationFilter, SecurityWebFiltersOrder.AUTHORIZATION)
-      .authorizeExchange(ex ->
-        ex
-          // Actuator endpoints for health checks
-          .pathMatchers("/actuator/health", "/actuator/metrics")
-          .permitAll()
-          // Leaderboard endpoints are public
-          .pathMatchers("/api/v1/leaderboards/**")
-          .permitAll()
-          // Everything else requires authentication
-          .anyExchange()
-          .authenticated()
-      )
-      .build();
+      // .oauth2ResourceServer(ServerHttpSecurity.OAuth2ResourceServerSpec::jwt) // enable if you use JWT
+      .authorizeExchange(exchanges -> {
+        // Always allow actuator probes
+        exchanges.pathMatchers("/actuator/health", "/actuator/metrics").permitAll();
+
+        // Guard: only register matchers when there's at least one
+        if (anonymousPaths.length > 0) {
+          exchanges.pathMatchers(anonymousPaths).permitAll();
+        }
```

Comment:
`pathMatchers` throws an error if it receives an empty array (i.e., if we have zero anonymously accessible paths). This code guards against defining `pathMatchers` if the input array is empty.